ex/auto/writer/odev_build_form/build_form.py

```diff
@@ -0,0 +1,447 @@
+        self.listen_to_text_field(props)
+
+        Forms.add_labelled_control(doc=BuildForm.doc, label="LASTNAME", comp_kind=Forms.CompenentKind.TextField, y=19)
+
+        props = Forms.add_labelled_control(
+            doc=BuildForm.doc, label="AGE", comp_kind=Forms.CompenentKind.NumericField, y=43
+        )
+        Props.set_property(props, "DecimalAccuracy", 0)
+
+        Forms.add_labelled_control(
+            doc=BuildForm.doc, label="BIRTHDATE", comp_kind=Forms.CompenentKind.FormattedField, y=51
+        )
+
+        # buttons, all with listeners
+        props = Forms.add_button(doc=BuildForm.doc, name="first", label="<<", x=2, y=63, width=8)
+        self.listen_to_button(props)
+
+        props = Forms.add_button(doc=BuildForm.doc, name="prev", label="<", x=12, y=63, width=8)
+        self.listen_to_button(props)
+
+        props = Forms.add_button(doc=BuildForm.doc, name="next", label=">", x=22, y=63, width=8)
+        self.listen_to_button(props)
+
+        props = Forms.add_button(doc=BuildForm.doc, name="last", label=">>", x=32, y=63, width=8)
+        self.listen_to_button(props)
+
+        props = Forms.add_button(doc=BuildForm.doc, name="new", label=">*", x=42, y=63, width=8)
+        self.listen_to_button(props)
+
+        props = Forms.add_button(doc=BuildForm.doc, name="reload", label="reload", x=58, y=63, width=13)
+        self.listen_to_button(props)
+        self.listen_to_mouse(props)
+
+        # some fixed text; no listener
+        Forms.add_control(
+            doc=BuildForm.doc,
+            name="text-1",
+            label="show only sales since",
+            comp_kind=Forms.CompenentKind.FixedText,
+            x=2,
+            y=80,
+            width=35,
+            height=6,
+        )
+
+        #  radio buttons inside a group box; use a property change listener
+        Forms.add_control(
+            doc=BuildForm.doc,
+            name="Options",
+            label="Options",
+            comp_kind=Forms.CompenentKind.GroupBox,
+            x=103,
+            y=5,
+            width=56,
+            height=25,
+        )
+
+        # these three radio buttons have the same name ("Option"), and
+        # so only one can be on at a time
+        props = Forms.add_control(
+            doc=BuildForm.doc,
+            name="Options",
+            label="No automatic generation",
+            comp_kind=Forms.CompenentKind.RadioButton,
+            x=106,
+            y=11,
+            width=50,
+            height=6,
+        )
+        props.addPropertyChangeListener("State", self)
+
+        props = Forms.add_control(
+            doc=BuildForm.doc,
+            name="Options",
+            label="Before inserting a record",
+            comp_kind=Forms.CompenentKind.RadioButton,
+            x=106,
+            y=17,
+            width=50,
+            height=6,
+        )
+        props.addPropertyChangeListener("State", self)
```

Comment:
Can this be set up with a Single Point of Reference? A comment line could be added when control first used to identify attributes used.

Something like the following with newly commented lines deleted:
```
        #  radio buttons inside a group box; use a property change listener
        Forms.add_control(
            doc=BuildForm.doc,
            name="Options",
            label="Options",
            comp_kind=Forms.CompenentKind.GroupBox,
            x=103,
            y=5,
            width=56,
            height=25,
        )

        # these three radio buttons have the same name ("Option"), and
        # so only one can be on at a time
        props = Forms.add_control(
            # doc=BuildForm.doc,
            # name="Options",
            label="No automatic generation",
            comp_kind=Forms.CompenentKind.RadioButton,
            x=106,
            # y=11,
            width=50,
            height=6,
            y=y + height,
            # possibly the following
            # indent=3,
            # x=x -indent,
            # width=width-2*indent,
        )
        props.addPropertyChangeListener("State", self)

        props = Forms.add_control(
            # doc=BuildForm.doc,
            # name="Options",
            label="Before inserting a record",
            # comp_kind=Forms.CompenentKind.RadioButton,
            # x=106,
            # y=17,
            y=y + height,
            # width=50,
            # height=6,
        )
        props.addPropertyChangeListener("State", self)

        props = Forms.add_control(
            # doc=BuildForm.doc,
            # name="Options",
            label="When moving to a new record",
            # comp_kind=Forms.CompenentKind.RadioButton,
            # x=106,
            # y=23,
            y=y + height,
            # width=50,
            # height=6,
        )
        props.addPropertyChangeListener("State", self)
```